lib/integrations/awsoidc/listdatabases_iam_config.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsoidc
+
+import (
+	"context"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/gravitational/trace"
+
+	awslib "github.com/gravitational/teleport/lib/cloud/aws"
+)
+
+const (
+	// defaultPolicyNameForListDatabases is the default name for the Inline Policy added to the IntegrationRole.
+	defaultPolicyNameForListDatabases = "ListDatabases"
```

Comment:
I don't think it is needed, or even a better approach.
This is part of the Role, which already has the teleport tags
```
// - teleport.dev/cluster: <clusterName>
// - teleport.dev/origin: aws-oidc-integration
// - teleport.dev/integration: <integrationName>
```

Assuming the user added `Teleport` somewhere in the role name, this would be repetitive.
`IntegrationRoleTeleport/TeleportListDatabases`

If this was a regular Policy, I would agree with you.
Given that this is an Inline Policy within a Role that was created specifically for this, I don't think this is necessary.